From 4e340b76026ffe564941bcc7cfa8f7d2f09b2bc8 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 3 Jul 2023 11:51:35 +0200
Subject: [PATCH] keyboard: Spawn "tecla" to show keyboard map

This is a modern replacement for gkbd-keyboard-display, stuck in gtk3
and X11 (libxklavier).
---
 panels/keyboard/cc-input-list-box.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/panels/keyboard/cc-input-list-box.c b/panels/keyboard/cc-input-list-box.c
index e32e0a7447..277c314001 100644
--- a/panels/keyboard/cc-input-list-box.c
+++ b/panels/keyboard/cc-input-list-box.c
@@ -239,10 +239,10 @@ row_layout_cb (CcInputListBox *self,
   layout_variant = cc_input_source_get_layout_variant (source);
 
   if (layout_variant && layout_variant[0])
-    commandline = g_strdup_printf ("gkbd-keyboard-display -l \"%s\t%s\"",
+    commandline = g_strdup_printf ("tecla \"%s+%s\"",
 				   layout, layout_variant);
   else
-    commandline = g_strdup_printf ("gkbd-keyboard-display -l %s",
+    commandline = g_strdup_printf ("tecla %s",
 				   layout);
 
   g_spawn_command_line_async (commandline, NULL);
-- 
GitLab

